js/hoverzoom.js

```diff
@@ -2643,12 +2639,27 @@ var hoverZoom = {
             });
         }
 
+        // get list of banned image, video or audio track urls
+        function loadBannedImages() {
+            chrome.runtime.sendMessage({action:'sendBannedImages'});
+        }
+
+        // check if url of image, video or audio track belongs to ban list
+        function isImageBanned(url) {
+            if (!url) return false;
+            reutrn !!bannedImagesList[url];
+        }
```

Comment:
@GrosPoulet Is `bannedImagesList` an array (list) or a set? According to https://stackoverflow.com/questions/2430000/determine-if-string-is-in-list-in-javascript if it's a list then you should use `.include` to check membership, and if it's a set then you should use `.has`.

Reply:
@extesy bannedImagesList is a set (key:value), urls are used as keys

Reply:
key:value is a dictionary (aka a map), not a set. Are values used for anything?

Reply:
value = location (page url) where banned image was found